exercises/binary-search/introduction.md

```diff
@@ -1,7 +1,7 @@
 # Introduction
 
 You have stumbled upon a group of mathematicians who are also singer-songwriters.
-They have written a song for each of their favorite numbers, and, as you can imagine, they have a lot of favorite numbers.
+They have written a song for each of their favorite numbers, and, as you can imagine, they have a lot of favorite numbers (like 0 or 73 or _pi_, which starts with 3.14).
```

Comment:
Wouldn't having pi as a favorite number imply that we expect the number to potentially be a floating-point number?

Reply:
Yeah, I was really hesitating about _pi_. Jeremy suggested it in the discussion on the Ruby track, and I like how it's a "famous" number. I also considered _e_ and _square root of -1_. And I don't like how 3.14 isn't pi it's just the start of it.

I picked 73 even though I don't watch Big Bang Theory, because Dr Sheldon is right about it being an awesome number.

> The best number is 73. Why? 73 is the 21st prime number. Its mirror, 37, is the 12th and its mirror, 21, is the product of multiplying 7 and 3.
> 
> In binary 73 is a palindrome, 1001001, which backwards is 1001001.

Jeremy also suggested 42, but that's not a math thing, it's a programmer/geek thing from the Hitchhiker's Guide to the Galaxy (which I tried reading and gave up on, so I'm biased against 42).

Do you have a suggestion for a nice number we can have instead of pi? I'd like a really big one that is interesting for mathematical reasons.

Reply:
```suggestion
They have written a song for each of their favorite numbers, and, as you can imagine, they have a lot of favorite numbers (like 0 or 73 or 6174).
```

Reply:
I like Kaprekar's number. The speed of light is more physics than maths.

Reply:
Yeah, speed of light is a nice one, though it isn't an abstract number (you'd need the unit in order to actually specify it. And also, as you say, physics, not maths).

Ok, cool. I'm going with Keprekar's constant!